src/parser/jobs/blm/modules/Leylines.tsx

```diff
@@ -0,0 +1,192 @@
+				}))
+			})
+		})
+
+		// Get the total duration of CoP uptime and Ley Lines, so we can get the overall percentage uptime
+		const copDuration = this.buffWindows[this.data.statuses.CIRCLE_OF_POWER.id].history.reduce((duration, cop) => duration + Math.max((cop.stop || 0) - cop.start, 0), 0)
+		const linesDuration = this.buffWindows[this.data.statuses.LEY_LINES.id].history.reduce((duration, lines) => duration + Math.max((lines.stop || 0) - lines.start, 0), 0)
+
+		this.checklist.add(new Rule({
+			name: <Trans id="blm.leylines.checklist-caption">Stay in your <ActionLink {...this.data.actions.LEY_LINES} /></Trans>,
+			description: <Trans id="blm.leylines.checklist">Try to avoid leaving your <ActionLink showIcon={false} {...this.data.actions.LEY_LINES} /> after placing them. Take advantage of <ActionLink showIcon={false} {...this.data.actions.LEY_LINES} />' size to stay in them while dodging AOEs and being in range of healers. If you can't stay in them for the majority of a <ActionLink showIcon={false} {...this.data.actions.LEY_LINES} />' duration, consider changing where they're placed in the fight.</Trans>,
+			requirements: [
+				new Requirement({
+					name: <ActionLink {...this.data.actions.LEY_LINES} />,
+					percent: this.dontMovePercent(copDuration, linesDuration),
+				}),
+			],
+			//pretty random. Should be revised, maybe based on fights? 10% is ~ 1 GCD. So we allow that.
+			target: 90,
+		}))
+	}
+
+	output() {
+		const fightEnd = this.parser.pull.timestamp + this.parser.pull.duration
+		return <Table collapsing unstackable compact="very">
+			<Table.Header>
+				<Table.Row>
+					<Table.HeaderCell><Trans id="blm.leylines.timestamp-header">Timestamp</Trans></Table.HeaderCell>
+					<Table.HeaderCell><Trans id="blm.leylines.uptime-header">Uptime</Trans></Table.HeaderCell>
+					<Table.HeaderCell></Table.HeaderCell>
+				</Table.Row>
+			</Table.Header>
+			<Table.Body>
+				{this.buffWindows[this.data.statuses.LEY_LINES.id].history.map(leyLinesEvent => {
+					// Find the CoPs that were inside this Ley Lines
+					const thisCoPHistory = this.buffWindows[this.data.statuses.CIRCLE_OF_POWER.id].history.filter(cops => ((cops.start >= leyLinesEvent.start) && ((cops.stop || 0) <= (leyLinesEvent.stop || 0))))
+
+					// For this set of CoPs, get the uptime
+					const thisCoPUptime = thisCoPHistory.reduce((duration, cop) => duration + Math.max((cop.stop || 0) - cop.start, 0), 0)
+
+					// Note that since we're getting the actual duration, rather than the expected duration,
+					// technically we'll call it 100% uptime if you stay in the lines and die halfway through...
+					// However, since that'll get flagged as a morbid checklist item, that's probably ok.
+					const thisPercent = this.dontMovePercent(thisCoPUptime, (leyLinesEvent.stop || fightEnd) - leyLinesEvent.start).toFixed(2)
+
+					return <Table.Row key={leyLinesEvent.start}>
+						<Table.Cell>{this.parser.formatTimestamp(leyLinesEvent.start)}</Table.Cell>
```

Comment:
```suggestion
						<Table.Cell>{this.parser.formatEpochTimestamp(leyLinesEvent.start)}</Table.Cell>
```

Analyzer system does some adjustments to zero under the hood, using the old formatTimestamp causes times to be wrong.

Reply:
Darn, been pretty good about catching those in other code. Fix't